### PR TITLE
Fixed identity operator evaluation and extended the accepted input types in compute_rdms

### PR DESCRIPTION
**Pull request by AlexandreF-1qbit**
There were several bugs preventing users to use the compute rdms function:
1.
 ```python
 {pauli_string_to_of(term): exp_val for term, exp_val in exp_data.items()}
``` 
to
```python
{pauli_string_to_of(term): exp_val for term, exp_val in exp_vals.items()}
``` 
(wrong dictionary).

2.
List are unhashable, changed output type of the `pauli_string_to_of` function to a tuple.

3. 
```python
if qterm in exp_vals:
    exp_val = exp_vals[qterm] if qterm else 1.
```
was failing for the `()` term. I added a conditional statement.

4. Can now accept dictionaries with string ("IXYZ") or openfermion-like ((1, "X), (2, "Y"), ...) terms.

### CODE REVIEW

File: tangelo/toolboxes/molecular_computation/rdms.py [hunk 1]
```diff
@@ -82,11 +81,16 @@ def compute_rdms(ferm_ham, mapping, up_then_down, exp_vals=None, exp_data=None,
         raise RuntimeError("Arguments exp_vals, exp_data and shadow are mutually exclusive. Provide exactly one of them.")
 
     # Initialize exp_vals
-    exp_vals = {pauli_string_to_of(term): exp_val for term, exp_val in exp_data.items()} \
-        if isinstance(exp_vals, dict) else {}
+    if isinstance(exp_vals, dict) and set(map(type, exp_vals)) == {str}:
```
Comment (ValentinS4t1qbit) [inline tangelo/toolboxes/molecular_computation/rdms.py:84]: A beautiful and compact way to use map to check all the keys in a dictionary for a given type. Thank you !

**Comment by AlexandreF-1qbit** [comment]
I tested all the possibilities, i.e.:
- `exp_data` with openfermion-like terms ((0, "X"), (1, "Y"), ...)
- `exp_data` with string terms ("XY...")
- `exp_vals` with openfermion-like terms
- `exp_vals` with string terms

Also, a missing entry in `exp_vals` raises an error. A missing entry in `exp_data` only raises an error when no compatible bases are detected.

**Comment by ValentinS4t1qbit** [comment]
Two things:

- Please change the PR title to something a bit better, that briefly describes what the PR did, for all readers.
- I would like you to add tests addressing the use cases this fix covers. The fact that all tests were passing before this PR and are still passing means that what you did is not reflected in them: we thus need more tests, that illustrate what you fixed / contributed.

**Review by ValentinS4t1qbit — APPROVED** [review]
Thank you !

**Pull request merged**